docs/release_process.md

```diff
@@ -0,0 +1,60 @@
+# cellxgene-schema Release Process
+
+_This document defines the release process of the pip installable `cellxgene-schema` tool to PyPI._
+
+The goal of the release process is to publish an installable package to PyPi, with a matching tagged release on github.
+
+## Overview
+
+The release process should result in the following side effects:
+
+* Version number bump, using semantic versioning
+* Tagged Github release
+* Publication to PyPi
+
+Note all release tags pushed to GitHub MUST follow semantic versioning.
+
+## How to release the next version of `cellxgene-schema`
+
+Steps must be run from the project directory and in a virtual env with all the dependencies defined in `requirements.txt` installed.
+
+1. Decide on whether the changes going into the next release warrant a patch, minor, or major version bump using [semantic versioning](https://semver.org/) and determine the version number based on the current latest version (i.e. if the current version is 2.0.3 and you'd like to do a minor version bump, then the next version will be 2.1.0).
+
+1. Create a release branch from latest main (e.g. `git checkout -b release-version-x.y.z`).
+
+1. In `setup.py` change the version number to be the version number determined in step 1.
+
+1. Update the [CHANGELOG](https://github.com/chanzuckerberg/single-cell-curation/blob/main/cellxgene_schema_cli/CHANGELOG.md) with the latest changes going into the new release. 
+
+1. Commit the changes, push the new branch to origin, and open a PR to be reviewed. Do not merge in the PR yet.
+
+1. Release the new version to Test PyPI. This can be done by running `make test-release`. Test the release by installing `cellxgene-schema` using pip `pip install -i https://test.pypi.org/simple/ cellxgene-schema` in a fresh virtual environment.
+
+1. If you find errors with the release candidate, fix them in `main`, delete the current release branch, and start over from step 1 with the next version number. Please also update the CHANGELOG with a note that the version number has been burned. This is necessary because once a version goes out on Test PyPI (or PyPI), that version number cannot be used again (i.e. there are no updates supported). For example, if you were trying to release a minor bump, say 2.1.0, and you found an error, then the next version will be 2.2.0 and 2.1.0 will be a "burn".
```

Comment:
To simplify the process, could this be done in the current working branch with an extra commit? 

Unless, is there a strong reason to start the process all over again?

Reply:
OK this is a bit moot now since with the release candidate, there is no longer any need to recreate a new branch. 

I would recommend that fixes typically be done in a separate branch anyways just so that the fix can actually go into `main` and not be confused with any changes that are related to version bumping. Small, atomic PRs are usually a best practice.